language/move-prover/bytecode/src/loop_analysis.rs

```diff
@@ -251,31 +251,40 @@ impl LoopAnalysisProcessor {
             builder.data.code[code_offset] = updated_goto;
         }
 
+        // we have unrolled the loop into a DAG, and there will be no loop invariants left
+        builder.data.loop_invariants.clear();
         builder.data
     }
 
     /// Collect invariants in the given loop header block
     ///
-    /// Loop invariants are defined as the longest sequence of consecutive 'assert'
-    /// statements in the loop header block, immediately after the Label statement.
+    /// Loop invariants are defined as the longest sequence of consecutive 'assert' statements in
```

Comment:
Shouldn't this comment say: ...longest sequence of consecutive `invariant` statements?

Reply:
My bad, yes, the longest sequence of consecutive `invariant` statements.

Reply:
Hmm, this is a bit tricky. Given that we don't have an `invariant` type in our `PropKind` (all we have is `assert`, `assume`, and `modifies`), the `invariant` type is simulated by the combination of
- the `PropKind::Assert` and
- its `AttrId` exists in `FunctionData.loop_invariants`, which is a set of `AttrId`.

I'll update the comment to be more explicit on this.